### PR TITLE
Improve search system for very large projects and other small fixes

### PR DESCRIPTION
**Pull request by tristanlatr**
Give a chance to each query to be aborted before posting a search to the worker. The search is aborted if the user keeps typing other words in a very short delay.

Simply disable the search as you type feature if index file is greater than 20MB. The typing experience in the input field was terrible, pressing enter is better.

### CODE REVIEW

**Comment by tristanlatr** [comment]
I've discovered some issues in the searching process, see https://github.com/yeraydiazdiaz/lunr.py/issues/112 but I think this PR can be merged before we this fix, then let's release a new version :)

File: docs/tests/test.py [hunk 1]
```diff
@@ -164,4 +167,44 @@ def test_search_index_generated():
...
     assert (BASE_DIR / 'api' / 'searchindex.json').is_file()
     assert (BASE_DIR / 'api' / 'fullsearchindex.json').is_file()
-    assert (BASE_DIR / 'api' / 'all-documents.html').is_file()
\ No newline at end of file
+    assert (BASE_DIR / 'api' / 'all-documents.html').is_file()
+
+def test_lunr_index() -> None:
+    """
+    Run some searches on the lunr index to test it's validity. 
+    There is currently an issue that prevent some results to be returned for specific keywords.
```
Comment (tristanlatr) [inline docs/tests/test.py:175]:
```suggestion

```

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

**Comment by tristanlatr** [comment]
Thanks!

**Pull request merged**